### PR TITLE
Set the rails logger after initialisation

### PR DESCRIPTION
**Pull request by iainbeeston**
## Goal

Currently, bugsnag logging does not work in rails unless you explicitly set the logger in your bugsnag initializer, and so all logging messages are being lost. I'd like to fix that so that it works out of the box.

### Changed

I've moved the code to set up the bugsnag logger to run after the railtie has been initialized. This fixes the issue for me (I believe because the rails logger has not been set up yet in the `before_initialize` hook, but is available in `after_initialize`).

## Tests

I think there are no existing tests for the railtie?

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: lib/bugsnag/integrations/railtie.rb [hunk 1]
```diff
@@ -22,7 +22,6 @@ class Railtie < Rails::Railtie
       # Skipping API key validation as the key may be set later in an
       # initializer. If not, the key will be validated in after_initialize.
       Bugsnag.configure(false) do |config|
-        config.logger = ::Rails.logger
```
Comment (tobyhs) [inline lib/bugsnag/integrations/railtie.rb]: I believe the logger was set here so that any log output from Bugsnag that occurred in a config/initializers/ file would end up in the Rails log.

When trying the example at https://github.com/bugsnag/bugsnag-ruby/tree/800ea83ee4885df2e2f2c48570e13ffa7e75b9b8/example/rails-51 with your change, Bugsnag log output resulting from a file in config/initializers/ doesn't end up in log/development.log.

**Comment by tomlongridge** [comment]
Closing, given the limitations noted above.
